aws/resource_aws_neptune_cluster_instance.go

```diff
@@ -0,0 +1,521 @@
+		createOpts.EngineVersion = aws.String(attr.(string))
+	}
+
+	if attr, ok := d.GetOk("preferred_backup_window"); ok {
+		createOpts.PreferredBackupWindow = aws.String(attr.(string))
+	}
+
+	if attr, ok := d.GetOk("preferred_maintenance_window"); ok {
+		createOpts.PreferredMaintenanceWindow = aws.String(attr.(string))
+	}
+
+	log.Printf("[DEBUG] Creating Neptune Instance: %s", createOpts)
+
+	var resp *neptune.CreateDBInstanceOutput
+	err := resource.Retry(1*time.Minute, func() *resource.RetryError {
+		var err error
+		resp, err = conn.CreateDBInstance(createOpts)
+		if err != nil {
+			if isAWSErr(err, "InvalidParameterValue", "IAM role ARN value is invalid or does not include the required permissions") {
+				return resource.RetryableError(err)
+			}
+			return resource.NonRetryableError(err)
+		}
+		return nil
+	})
+	if err != nil {
+		return fmt.Errorf("error creating Neptune Instance: %s", err)
+	}
+
+	d.SetId(aws.StringValue(resp.DBInstance.DBInstanceIdentifier))
+
+	stateConf := &resource.StateChangeConf{
+		Pending:    resourceAwsNeptuneClusterInstanceCreateUpdatePendingStates,
+		Target:     []string{"available"},
+		Refresh:    resourceAwsNeptuneInstanceStateRefreshFunc(d.Id(), conn),
+		Timeout:    d.Timeout(schema.TimeoutCreate),
+		MinTimeout: 10 * time.Second,
+		Delay:      30 * time.Second,
+	}
+
+	// Wait, catching any errors
+	_, err = stateConf.WaitForState()
+	if err != nil {
+		return err
+	}
+
+	return resourceAwsNeptuneClusterInstanceRead(d, meta)
+}
+
+func resourceAwsNeptuneClusterInstanceRead(d *schema.ResourceData, meta interface{}) error {
+	db, err := resourceAwsNeptuneInstanceRetrieve(d.Id(), meta.(*AWSClient).neptuneconn)
+	if err != nil {
+		return fmt.Errorf("Error on retrieving Neptune Cluster Instance (%s): %s", d.Id(), err)
+	}
+
+	if db == nil {
+		log.Printf("[WARN] Neptune Cluster Instance (%s): not found, removing from state.", d.Id())
+		d.SetId("")
+		return nil
+	}
+
+	if db.DBClusterIdentifier == nil {
+		return fmt.Errorf("Cluster identifier is missing from instance (%s)", d.Id())
+	}
+
+	conn := meta.(*AWSClient).neptuneconn
+	resp, err := conn.DescribeDBClusters(&neptune.DescribeDBClustersInput{
+		DBClusterIdentifier: db.DBClusterIdentifier,
+	})
+
+	var dbc *neptune.DBCluster
+	for _, c := range resp.DBClusters {
+		if aws.StringValue(c.DBClusterIdentifier) == aws.StringValue(db.DBClusterIdentifier) {
+			dbc = c
+		}
+	}
+	if dbc == nil {
+		return fmt.Errorf("Error finding Neptune Cluster (%s) for Cluster Instance (%s): %s",
+			aws.StringValue(db.DBClusterIdentifier), aws.StringValue(db.DBInstanceIdentifier), err)
+	}
+	for _, m := range dbc.DBClusterMembers {
+		if aws.StringValue(db.DBInstanceIdentifier) == aws.StringValue(m.DBInstanceIdentifier) {
+			if aws.BoolValue(m.IsClusterWriter) == true {
+				d.Set("writer", true)
+			} else {
+				d.Set("writer", false)
+			}
+		}
+	}
+
+	if db.Endpoint != nil {
+		d.Set("endpoint", db.Endpoint.Address)
```

Comment:
Nitpick: with the `aws_db_instance` resource we have 3 attributes (`address`, `endpoint`, and `port`):

https://github.com/terraform-providers/terraform-provider-aws/blob/2c5c645f5a088c38ac490f032adaf1706e4c09a7/aws/resource_aws_db_instance.go#L987-L995

For a few reasons I think we should follow suit here:
* Provides an attribute with only the hostname
* Matches the API naming for address
* Ancillary: Reduces any confusion working between the two resources